shapely/ops.py

```diff
@@ -121,37 +122,15 @@ def cascaded_union(self, geoms):
 
         This method was superseded by :meth:`unary_union`.
         """
-        try:
-            L = len(geoms)
-            if isinstance(geoms, BaseMultipartGeometry):
-                geoms = geoms.geoms
-        except TypeError:
-            geoms = [geoms]
-            L = 1
-        subs = (c_void_p * L)()
-        for i, g in enumerate(geoms):
-            subs[i] = g._geom
-        collection = lgeos.GEOSGeom_createCollection(6, subs, L)
-        return geom_factory(lgeos.methods['cascaded_union'](collection))
+        return pygeos.union_all(geoms, axis=None)
 
     def unary_union(self, geoms):
         """Returns the union of a sequence of geometries
 
         This method replaces :meth:`cascaded_union` as the
         prefered method for dissolving many polygons.
         """
-        try:
-            L = len(geoms)
-            if isinstance(geoms, BaseMultipartGeometry):
-                geoms = geoms.geoms
-        except TypeError:
-            geoms = [geoms]
-            L = 1
-        subs = (c_void_p * L)()
-        for i, g in enumerate(geoms):
-            subs[i] = g._geom
-        collection = lgeos.GEOSGeom_createCollection(6, subs, L)
-        return geom_factory(lgeos.methods['unary_union'](collection))
+        return pygeos.union_all(geoms, axis=None)
```

Comment:
similar to comment above, is pygeos calling `GEOSUnaryUnion_r` here?

Reply:
> similar to comment above, is pygeos calling GEOSUnaryUnion_r here?

Yes, it's a potential naming issue on the pygeos side, but `union_all` is basically `unary_union` (the main difference being that it supports an `axis` keyword in case you have a multidimensional array to union along a specific axis). But we should certainly consider addng a `unary_union` function as well to be compatible with Shapely's current naming.

Reply:
Relevant issue about `union_all` vs `unary_union` is https://github.com/pygeos/pygeos/issues/194